huffman/src/symbol_span.hpp

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include "huffman/src/utility.hpp"
+
+#include <concepts>
+#include <ranges>
+
+namespace starflate::huffman {
+
+/// An inclusive span of symbols
+///
+template <symbol S>
+class symbol_span : public std::ranges::view_interface<symbol_span<S>>
+{
+  S first_;
+  S last_;
+
+  using range_type = std::ranges::iota_view<S, S>;
+
+  constexpr auto as_range() const -> range_type
+  {
+    auto tmp = last_;
```

Comment:
`iota_view` excludes `last_` so we want `last_ + 1`.

But if `last_` is `char`, `last_ + 1` will promote to `int` . `++tmp` does not change the type - it can't since it returns a reference to `tmp`.

Reply:
Ugh.
Can we do?
`range_type{first_, last_ + static_cast<S>(1)}`
That seems to be a bit more obvious to me

Reply:
Sure let's do that. You'll need a cast on the addition result due to interger promotion.

I'm also on the trainer atm so feel free to make any changes.